src/state_machine.zig

```diff
@@ -521,6 +521,85 @@ pub fn StateMachineType(
             }
         };
 
+        const Metrics = struct {
+            const TimingSummary = struct {
+                duration_min_us: ?u64 = null,
+                duration_max_us: ?u64 = null,
+                duration_sum_us: u64 = 0,
+
+                count: u64 = 0,
+
+                /// If an operation supports batching (eg, lookup_accounts) this is a count of the
+                /// number of internal operations.
+                count_batch: u64 = 0,
+            };
+
+            /// Technically 'timer' can't be used, but that'll error out at comptime.
+            const MetricEnum = std.meta.FieldEnum(Metrics);
+
+            create_accounts: TimingSummary = .{},
+            create_transfers: TimingSummary = .{},
+            lookup_accounts: TimingSummary = .{},
```

Comment:
IIUC, all of these are currently measuring the time taken across `prefetch`, `reply_setup`, and `commit`, where `reply_setup` is a VSR-level operation, not state machine level. 

Would it make more sense to measure timings across (`prefetch` → `prefetch_finish`), and within `commit` separately? Or do we want to club `prefetch` & `commit` because `commit` is purely in-memory?

Reply:
Both of those are good points! 

I think adding more granularity (either in a followup, or for the real metrics work) will be the way to go.

Reply:
Ack, sounds reasonable, we can take these up as enhancements in the current metrics in case we need more granularity. 👍🏼 

Those were the only questions I had, apart from the that this looks great to me. Approving!